### PR TITLE
getInterfaceType() always returns a type

### PR DESCRIPTION
**Pull request by hamishknight**
Remove a handful of checks for whether `getInterfaceType()` returns a non-null type, as it now always returns a non-null type. Also remove a couple of unnecessary calls to `hasInterfaceType()`.

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test

File: lib/Sema/TypeCheckStmt.cpp [hunk 1]
```diff
@@ -1166,8 +1166,7 @@ class StmtChecker : public StmtVisitor<StmtChecker, Stmt*> {
         }
         assert(isa<CaseStmt>(initialCaseVarDecl->getParentPatternStmt()));
 
-        if (vd->getInterfaceType() && initialCaseVarDecl->getType() &&
```
Comment (CodaFi) [inline lib/Sema/TypeCheckStmt.cpp]: #VarDecl::getType() also computes the interface type.

Comment (hamishknight) [reply lib/Sema/TypeCheckStmt.cpp]: Ah nice, even more cleanup

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test

**Review by slavapestov — APPROVED** [review]
Excellent! There are various radars floating around where getInterfaceType() returns null and the caller doesn't expect it, great that they're all fixed now.

**Comment by CodaFi** [comment]
This picked up the libSyntax failures

@swift-ci please test macOS platform

**Pull request merged**